src/Razor/src/Microsoft.AspNetCore.Razor.ExternalAccess.RoslynWorkspace/RazorWorkspaceListener.cs

```diff
@@ -45,17 +50,38 @@ public void Dispose()
         _disposeTokenSource.Dispose();
     }
 
-    public void EnsureInitialized(Workspace workspace, string projectInfoFileName)
+    public void EnsureInitialized(Workspace workspace, string pipeName)
+    {
+        // Make sure we don't hook up the event handler multiple times
+        if (_stream is not null)
+        {
+            return;
+        }
+
+        _logger.LogTrace("Opening named pipe server: {0}", pipeName);
```

Comment:
Do all the processes write to a single log stream or does each one write to a different one? If different then strongly suggest you include the process id in any named pipe logging. Will save you hair pulling exercises in the future when trying to figure out which process is doing what :smile:

Reply:
In VS Code this goes to the vs code output window, so I think we'll be okay. We don't write to a file on disk :)